1/backend/src/resolvers.js

```diff
@@ -1,67 +1,268 @@
+                    json_data = JSON.parse(JSON.stringify(todos))
+                    return json_data
+                } finally {
+                    await session.close()
+                }
+            }
+        },
+        completedTodos: async (parent, args, context) => {
+            const {driver} = context
+            const getCompletedTodosCypher = `
+                    MATCH (todo:Todo) WHERE todo.completed = true 
+                    RETURN todo.id, todo.message, todo.completed ORDER BY todo.message
+             `
+            const session = driver.session()
+            try {
+                data = await session.run(getCompletedTodosCypher)
+                const todos = await data.records.map(record => ({
+                    id: record.get('todo.id'),
+                    message: record.get('todo.message'),
+                    completed: record.get('todo.completed')
+
+                }))
+                json_data = JSON.parse(JSON.stringify(todos))
+                return json_data
+            } finally {
+                await session.close()
+            }
+        }
     },
     Mutation: {
-        addTodo: (parent, args, context) => {
-            if(args.message != "" && args.message != null && context.token) {
-                let newTodo = {
-                    id: findNextId(),
+        addTodo: async (parent, args, context) => {
+            if (args.message != "" && args.message != null && context.token) {
+                const {driver} = context
+                const createTodoCypher = `
+                    CREATE (todo:Todo {params})
+                    RETURN todo.id, todo.message, todo.completed
+                `
+                const params = {
                     message: args.message,
                     completed: false,
-                };
-                data.todos.push(newTodo);
-                return newTodo;
+                    id: uuid()
+                }
+                const session = driver.session()
+                try {
+                    data = await session.run(createTodoCypher, {params})
+                    const [todo] = await data.records.map(record => ({
+                        id: record.get('todo.id'),
+                        message: record.get('todo.message'),
+                        completed: record.get('todo.completed')
+
+                    }))
+                    return todo;
+                } finally {
+                    await session.close()
+                }
             }
             return;
         },
-        deleteTodo: (parent, args, context) => {
-            if (args.id != null && context.token){
-                let index = data.todos.findIndex(td => td.id === parseInt(args.id));
-                if (index > 0){
-                    data.todos.splice( index, 1)
+        addUser: async (parent, args, context) => {
+            if (args.login != "" && args.login != null && args.password != "" && args.password != null) {
```

Comment:
Throw `ArgumentError` in the `else` block. You want to validate here, right? Even better, you could write `password: String!` in your graphql type definition and enforce a non-null value.